[KB] SAML - Add auth method mismatch + consistency, procedural, language improvements (#1517)
semgrep/semgrep-docs: docs/kb/semgrep-appsec-platform/saml-stops-working.md

import MoreHelp from "/src/components/MoreHelp"

# Troubleshooting SAML SSO

This article walks you through troubleshooting SAML SSO failures, including the case where your SAML configuration stops working after you've successfully configured it and used it for some time. There are several common reasons why a configuration may fail.

## Certificate issues

> If you see the error `Signature validation failed. SAML Response rejected` it's likely there is a problem with the certificate. If the IdP certificate becomes invalid via expiration, revocation, or rotation, signature validation fails.

Reviewer: ```suggestion
If you see the error `Signature validation failed. SAML Response rejected` it's likely there is a problem with the certificate. The IdP certificate may be invalid due to expiration, revocation, or rotation, or signature validation fails.
```

Author: This is not semantically the same as what I said - the signature validation failure is a result of the other issues, not a cause of an issue. Skipping this one, feel free to adjust the wording later as long as the sense is preserved.

Author: That's probably fine for this context. It fails if the cert becomes invalid for any reason, but those are the most common reasons. I'll make the change.
